aibolit/metrics/cognitiveC/cognitive_c.py

```diff
@@ -1,130 +1,139 @@
+    def _check_if_statement(self, ast, expr, nested_level: int) -> None:
         '''function to work with IfStatement block'''
-        self._get_complexity(expr.condition, 0)
-        if expr.then_statement is not None:
+        all_childs = [i for i in ast.tree.succ[expr]]
+        self._get_complexity(ast, all_childs[0], 0)
+        if len(all_childs) >= 2:
             self.complexity += nested_level + 1
-            self._get_complexity(expr.then_statement, nested_level + 1)
+            self._get_complexity(ast, all_childs[1], nested_level + 1)
 
-        if expr.else_statement is not None:
-            if isinstance(expr.else_statement, IfStatement):
+        if len(all_childs) == 3:
+            if ast.get_type(all_childs[2]) == ASTNodeType.IF_STATEMENT:
                 self.complexity -= nested_level
-                self._check_if_statement(expr.else_statement, nested_level)
+                self._check_if_statement(ast, all_childs[2], nested_level)
             else:
                 self.complexity += 1
-                self._get_complexity(expr.else_statement, nested_level + 1)
+                self._get_complexity(ast, all_childs[2], nested_level + 1)
 
-    def _increment_logical_operators(self, block: BinaryOperation, prev_operator: str) -> None:
-        for each_block in [block.operandr, block.operandl]:
+    def _increment_logical_operators(self, ast: AST, binary_operation_node: int) -> None:
+        logical_operators_sequence = self._create_logical_operators_sequence(ast, binary_operation_node)
+        self.complexity += len(list(groupby(logical_operators_sequence)))
 
-            if isinstance(each_block, BinaryOperation) and prev_operator in logical_operators:
-                if prev_operator != each_block.operator:
-                    self.complexity += 1
+    def _create_logical_operators_sequence(self, ast: AST, binary_operation_node: int) -> List[str]:
+        if ast.get_type(binary_operation_node) != ASTNodeType.BINARY_OPERATION:
+            return []
 
-                elif isinstance(each_block.operandr, BinaryOperation):
-                    self.complexity += 1
+        operator, left_side_node, right_side_node = ast.get_binary_operation_params(binary_operation_node)
+        if operator not in logical_operators:
+            return []
 
-                self._increment_logical_operators(each_block, each_block.operator)
+        left_sequence = self._create_logical_operators_sequence(ast, left_side_node)
+        right_sequence = self._create_logical_operators_sequence(ast, right_side_node)
+        return left_sequence + [operator] + right_sequence
 
-    def _is_recursion_call(self, block: MethodInvocation) -> bool:
-        if self.method_name == block.member:
-            return True
-        return False
+    def _is_recursion_call(self, ast, node) -> bool:   # type: ignore
+        if ast.get_type(node) == ASTNodeType.METHOD_INVOCATION:
```

Comment:
We should do one of the following thing:

- `assert` node to be `ASTNodeType.METHOD_INVOCATION` type
- Make method return something in all `if/else` branches

Reply:
used assert